### ISSUE TITLE
Freezing IPython version 4 causes recursion limit exceeded

### ISSUE THREAD

**Issue by anthony-tuininga** [marcelotduarte/cx_Freeze#180]
Originally reported by: **R KH (Bitbucket: [rkh_](https://bitbucket.org/rkh_), GitHub: Unknown)**

----------------------------------------

With cx_freeze 4.3.3 and IPython version 4.0.3, attempting to freeze with IPython included causes a recursive inclusion of some modules:

```
#!python


  File "/usr/local/lib/python2.7/dist-packages/cx_Freeze-4.3.3-py2.7-linux-x86_64.egg/cx_Freeze/finder.py", line 504, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "/usr/local/lib/python2.7/dist-packages/cx_Freeze-4.3.3-py2.7-linux-x86_64.egg/cx_Freeze/finder.py", line 603, in _ScanCode
    module, relativeImportIndex)
  File "/usr/local/lib/python2.7/dist-packages/cx_Freeze-4.3.3-py2.7-linux-x86_64.egg/cx_Freeze/finder.py", line 358, in _ImportModule
    deferredImports, namespace = namespace)
  File "/usr/local/lib/python2.7/dist-packages/cx_Freeze-4.3.3-py2.7-linux-x86_64.egg/cx_Freeze/finder.py", line 438, in _InternalImportModule
    parentModule, namespace)
  File "/usr/local/lib/python2.7/dist-packages/cx_Freeze-4.3.3-py2.7-linux-x86_64.egg/cx_Freeze/finder.py", line 451, in _LoadModule
    module = self._AddModule(name)
  File "/usr/local/lib/python2.7/dist-packages/cx_Freeze-4.3.3-py2.7-linux-x86_64.egg/cx_Freeze/finder.py", line 215, in _AddModule
    module = self._modules[name] = Module(name)

```

The stack trace repeats of course.

Adding the following in the setup.py for the freeze:

```
#!python

try:
    import IPython
    ipython_major, ipython_minor, ipython_rev = IPython.version_info[:3]
    print "IPython version", ipython_major
    if ipython_major in (4,):
        additonal_excludes += ['traitlets.traitlets',
                               'collections.sys',
                               'collections._weakref', ]
        print "fixing IPython module exclusions", ipython_major
except Exception as e:
    logging.warning("trouble setting up IPython excludes", exc_info=e)
```
And adding the "additonal_excludes" to the excludes section of the freeze options fixes the "recursion limit exceeded" error and allows the freeze to complete.  My app runs normally, but I am not sure if there is still a path that will actually need those modules...

----------------------------------------
- Bitbucket: https://bitbucket.org/anthony_tuininga/cx_freeze/issue/180

**Comment by anthony-tuininga** [comment]
*Original comment by* **Anthony Tuininga (Bitbucket: [anthony_tuininga](https://bitbucket.org/anthony_tuininga), GitHub: Unknown)**:

----------------------------------------

I believe this issue has been corrected (see issue #127). Can you confirm? Thanks.

**Comment by marcelotduarte** [comment]
cx_Freeze 6.6 has just been released.
Closing due to inactivity. 
If you had issues please open a new issue.
